components/addFileButton/index.jsx

```diff
@@ -19,48 +19,195 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import { useContext, useState } from "react";
+import { useContext, useState, useEffect } from "react";
 import T from "prop-types";
-import { saveFileInContainer } from "@inrupt/solid-client";
+import { overwriteFile, fetchResourceInfoWithAcl } from "@inrupt/solid-client";
 import { Button } from "@material-ui/core";
-import PodLocationContext from "../../src/contexts/podLocationContext";
 import SessionContext from "../../src/contexts/sessionContext";
+import PodLocationContext from "../../src/contexts/podLocationContext";
 import AlertContext from "../../src/contexts/alertContext";
+import ConfirmationDialogContext from "../../src/contexts/confirmationDialogContext";
+
+export function handleSaveResource({
```

Comment:
If I understand correctly, these methods are exposed only so that you can test them, right? Might want to add

```
/**
 * @internal
 */
```

to indicate that they are not part of any public API.

Reply:
I don't think we use typedoc anywhere - throughout the codebase we're exporting little functions like these. I don't think it's particularly necessary to call out as internal.

Reply:
True, and it's not like we're exposing PodBrowser as a library, so no need to indicate internal and public API. So ignore my previous suggestion ^_^